vcd/resource_vcd_vapp_vm.go

```diff
@@ -309,15 +315,32 @@ func resourceVcdVAppVmCreate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	// for back compatibility we allow to set computer name from `name` if computer_name isn't provided
+	var computerName string
+	if cName, ok := d.GetOk("computer_name"); ok {
+		computerName = cName.(string)
+	} else {
+		computerName = d.Get("name").(string)
+	}
+
 	if initScript, ok := d.GetOk("initscript"); ok {
-		task, err := vm.RunCustomizationScript(d.Get("name").(string), initScript.(string))
+		task, err := vm.RunCustomizationScript(computerName, initScript.(string))
 		if err != nil {
 			return fmt.Errorf("error with init script setting: %#v", err)
 		}
 		err = task.WaitTaskCompletion()
 		if err != nil {
 			return fmt.Errorf(errorCompletingTask, err)
 		}
+	} else {
+		task, err := vm.Customize(computerName, "", false)
+		if err != nil {
+			return fmt.Errorf("error with applying computer name: %#v", err)
+		}
+		err = task.WaitTaskCompletion()
+		if err != nil {
+			return fmt.Errorf(errorCompletingTask, err)
+		}
```

Comment:
This comment is about the whole 318:343 approach. As it is right now, we're mixing the old behaviour with the new one, but I believe we should be **prioritising new behaviour** and leaving the old logic intact. In other words, we can simplify as follows:

1. Leave old code intact, only add a warning message inside it:

```
	if initScript, ok := d.GetOk("initscript"); ok {
			task, err := vm.RunCustomizationScript(d.Get("name").(string), initScript.(string))
			if err != nil {
				return fmt.Errorf("error with init script setting: %#v", err)
			}
			err = task.WaitTaskCompletion()
			if err != nil {
				return fmt.Errorf(errorCompletingTask, err)
			}
```

2. Remove "else" and just do this part if the new `computer_name` is defined:

```
	} else { // Remove and check for computer_name instead
			task, err := vm.Customize(computerName, "", false)
			if err != nil {
				return fmt.Errorf("error with applying computer name: %#v", err)
			}
			err = task.WaitTaskCompletion()
			if err != nil {
				return fmt.Errorf(errorCompletingTask, err)
			}
```

So, if the user defines the new `computer_name` we don't care about old way of setting it. If they don't, we use the old code without modifications and print a warning.

CC: @Didainius @dataclouder

Reply:
Don't forget we need to leave `initscript` intact! And with your proposal we would overwrite `initscript` to "".

Reply:
No, because:
"} else { // Remove and check for computer_name instead"
Idea is that you set computer_name only if it's defined. If it's undefined, you don't "overwrite" the old logic.

Reply:
With current API, to change computer_name you have to provide initscript - what's why it's implemented as it is now. In general all logic is left - no previous behaviour is removed - it works as it was working.

Reply:
I think @lvirbalas is right. The `else` block will always be triggered now and VMname will always be set.

Reply:
Yep, for this case I added change